### DISCUSSION TITLE
Website Using AspNetStatic

### DISCUSSION THREAD

**Discussion by timalmond** [ZarehD/AspNetStatic#13] [Show and tell]
Hi. I just thought I'd post about how I've used AspNetStatic to allow my website to run as a static site on Netlify. Please ask any questions

https://timalmond.com/blog/anatomy-of-a-website

**Comment by ZarehD** [comment]
Nice work, Tim. Thank you ;-)

**Comment by MrninOFF** [comment]
Hi, your post motivated me to finally create an all about me website to track my future projects and life goals. I was a little confused about how the StaticResourcesInfoProvider works, but in about 15 minutes I had it mapped and running for a default generated .net website project. The website is hosted here: mrnin.dev (GitHub site)

**Reply by ZarehD** [reply]
Hi, @MrninOFF. I'm glad you found AspNetStatic useful. Was there something in particular about `StaticResourcesInfoProvider` that you found confusing? Just curious; maybe there's something I can do to make it easier.

**Reply by MrninOFF** [reply]
I am not sure if u are able to do it or if it would be a security problem, but after u generate all *.html files, u could simple go file by file and read <header and <script. After that u would find all files (css, js) needs to be copies and user would not need to manualy define a road to it. Same could go for images, all u need is one or two simple regex conditions.

**Reply by ZarehD** [reply]
That's true; it could also enumerate all the .cshtml files in the project and generate static files for them. It could do the same for the wwwroot folder contents. This may seem convenient at first, but I think it would make the module very opinionated about the project content, and would result in the need for a good bit of additional settings to let users alter or override that behavior. I tend to favor letting users be explicit about what they want; it's much more straightforward that way.

That said, there's no reason why a user can't write a bit of code to enumerate & parse their project's files (based on some unique logic) in order to create the collection that is passed to StaticResourcesInfoProvider.
